backend/src/core/models/steam.app.js

```diff
@@ -98,4 +66,70 @@ export class SteamApp {
 
     return steamAppsSource.filter((app) => !targetAppIds.includes(app.appid));
   }
+
+  get appType() {
+    return this.type;
+  }
+
+  set appType(type) {
+    this.type = type;
+  }
+
+  isGame() {
+    return this.type === SteamApp.validTypes.game;
+  }
+
+  recordHtmlAttempt(page, source) {
+    this.addHtmlPageTriedViaSource(source);
+
+    if (page.toString() === "") this.addHtmlPageFailedViaSource(source);
+  }
+
+  addHtmlPageTriedViaSource(source) {
+    if (this.triedVia.includes(source)) return;
+
+    this.triedVia.push(source);
+  }
+
+  addHtmlPageFailedViaSource(source) {
+    if (this.failedVia.includes(source)) return;
+
+    this.failedVia.push(source);
+  }
+
+  updateSteamAppType(page, source) {
+    this.appType = this.#getType(page, source);
+  }
+
+  #getType(page, source) {
+    if (source === ValidDataSources.validDataSources.steamWeb)
+      return this.#getSteamWebAppType(page);
+    if (source === ValidDataSources.validDataSources.steamcharts)
+      return this.#getSteamchartsAppType(page);
+  }
+
+  #getSteamWebAppType(page) {
```

Comment:
ok, very good job moving it here AND encapsulate it by making the method private :+1: